### PR TITLE
add lognormal

### PR DESCRIPTION
**Pull request by erindb**
#910

### CODE REVIEW

**Comment by null-a** [comment]
This is great! The sampler, scorer, tests, docs, all look good to me. I can see just one subtle issue that needs addressing before this is merged. When using gradient based inference the parameters of a distribution can be AD values rather than regular JS numbers, which at present the logNormal sample doesn't handle. For example, the following program will generate an error at present:

```js
var model = function() {
  return logNormal(0, 1);
};

Infer({model, method: 'optimize'});
```

Thankfully the fix is easy, you just need to use `ad.value(...)` to get the underlying values out of the AD objects before calling the sampling routine. I'll add a comment inline to indicate where this ought to happen.

It would also be good to have a test for this. The easiest way to achieve this is by adding a new inference test. You'd need to add two new files (one for the model and one to describe the expected marginal) and also an entry to the 'OptimizeELBO' section of `test-inference.js`. Something analogous to this exponential test would work:

https://github.com/null-a/webppl/blob/dev/tests/test-data/stochastic/models/exponential.wppl
https://github.com/null-a/webppl/blob/dev/tests/test-data/stochastic/expected/exponential.json
https://github.com/null-a/webppl/blob/c757f25adfc99bf0da88d02987beb26460974703/tests/test-inference.js#L666

One final thing, it would be great to add reparameterisation to this at some point. If you're interesting in doing that here or as a follow-up and could use some pointers I'd be happy to help.

File: src/dists/logNormal.ad.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+var LogNormal = base.makeDistributionType({
+  name: 'LogNormal',
+  desc: 'Distribution over non-negative reals where the log of the random ' +
+    'variable is normally distributed.',
+  params: [{name: 'mu', desc: 'mean', type: types.unboundedReal},
+           {name: 'sigma', desc: 'standard deviation', type: types.positiveReal}],
+  wikipedia: 'Log-normal_distribution',
+  mixins: [base.continuousSupport],
+  sample: function() {
+    return sample(this.params.mu, this.params.sigma);
```
Comment (null-a) [inline src/dists/logNormal.ad.js]:
```suggestion
    return sample(ad.value(this.params.mu), ad.value(this.params.sigma));
```

`ad.value(...)` get the under lying value out of an AD node. If its argument is a number, it acts like the identity. This should allow the example in my comment to run without error.

**Comment by null-a** [comment]
@erindb Thanks for adding the `ad.value` fix.

**Comment by erindb** [comment]
Thanks! Sorry I never got around to the other stuff! :)

**Pull request merged**